src/dune_engine/sandbox.ml

```diff
@@ -41,6 +41,22 @@ type t =
 let dir t = t.dir
 let map_path t p = Path.Build.append t.dir p
 
+let rec copy_recursively (src_kind : Unix.file_kind) ~src ~dst =
```

Comment:
We have something similar to this in `Pkg_rules.Copy_tree`. Would be good to share implementations one day.

Reply:
I tried to factor with `collect_dir_recursively` too but it didn't go well.